src/kafka-util/src/bin/kgen.rs

```diff
@@ -437,208 +440,224 @@ impl<'a> ValueGenerator<'a> {
     }
 }
 
+arg_enum! {
+    pub enum KeyFormat {
```

Comment:
this was true before as well so not a big deal at all but this categorization conflates format (which is really avro or unsigned int/bytes) with the distribution of the integer keys. I would be in favor of just getting rid of sequential -- there isn't really a analogue for sequential value distribution and idk what a compelling use case for sequential key generation over uniform sampling would be

Reply:
if someone has a compelling reason though I'm all for just keeping it the way it is:

I will say that the partition_round_robin flag makes it so that if you want, you can get a totally even spread of messages to partitions. The only other use case I can think of for sequential is "I want all of my keys to be unique", because even sampling over 64 bits, your collision probability after sending N messages grows exponentially with N iirc